src/main/kotlin/com/chattriggers/ctjs/api/message/TextComponent.kt

```diff
@@ -190,9 +195,14 @@ class TextComponent : Text {
      * Sets the value to be used by the hover action. The value is interpreted according to [hoverAction]
      */
     fun setHoverValue(value: Any?) = apply {
-        hoverValue = if (hoverAction == HoverEvent.Action.SHOW_TEXT && value != null) {
-            from(value)
-        } else value
+        hoverValue = value?.let {
+            when (hoverAction) {
+                HoverEvent.Action.SHOW_TEXT -> from(value)
+                HoverEvent.Action.SHOW_ITEM -> parseItemContent(value)
+                HoverEvent.Action.SHOW_ENTITY -> parseEntityContent(value)
+                else -> value
```

Comment:
If the only other case is `null`, lets write that explicitly instead of using `else`. That way if another enum is added in a future version, it'll be a compiler error.

Reply:
The only other case is not `null` due to `hoverAction` being of type `HoverEvent.Action<*>?`

Reply:
Ahh is this the class that isn't an enum? In that case nvm